src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets

```diff
@@ -374,11 +374,16 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
     <Warning Text="Unable to resolve a Razor configuration for the value '$(RazorDefaultConfiguration)'. Available configurations '@(RazorConfiguration->'%(Identity)', ', ')'.'"
       Code="RAZORSDK1000"
-      Condition="'@(ResolvedRazorConfiguration->Count())' == '0'" />
+      Condition="'$(_Targeting30OrNewerRazorLangVersion)' == 'true' AND '@(ResolvedRazorConfiguration->Count())' == '0'" />
 
     <Warning Text="More than one Razor configuration was resolved for the value '$(RazorDefaultConfiguration)'. Available configurations '@(RazorConfiguration->'%(Identity)', ', ')'."
       Code="RAZORSDK1001"
-      Condition="'@(ResolvedRazorConfiguration->Count())' &gt; '1'" />
+      Condition="'$(_Targeting30OrNewerRazorLangVersion)' == 'true' AND '@(ResolvedRazorConfiguration->Count())' &gt; '1'" />
```

Comment:
I think you answered this in your other comment. Can you expand on 

> You get an unnecessary warning in that case.

Aka, what's the warning, in VS only? Outside of VS?

Reply:
You get the text of this warning, because essentially you get `RazorLangVersion` = 2.1 (the default), `RazorDefaultConfiguration`= `<empty>` and consequently no `RazorConfiguration` items are resolved.

Reply:
And it's specifically in VS, for design time builds, during \ prior to restore. It goes away on rebuilds.